### ISSUE TITLE
using the middleware with rails 8 breaks rails routes generation

### ISSUE THREAD

**Issue by francescob** [railsware/js-routes#319]
I've upgraded my application to rails 8 and now the line 
`  config.middleware.use(JsRoutes::Middleware)`

in development.rb causes the rails routes to be empty. Removing the line the application works as normal

**Comment by le0pard** [comment]
try

```ruby
Rails.application.routes_reloader.try(:execute_unless_loaded)
```

before 

```ruby
config.middleware.use(JsRoutes::Middleware)
```

**Comment by francescob** [comment]
thanks for the super quick reply, unfortunately that line leads to the app failing to boot with the following error:

`/home/francesco/.asdf/installs/ruby/3.3.4/lib64/ruby/gems/3.3.0/gems/railties-8.0.0/lib/rails/engine.rb:602:in `unshift': can't modify frozen Array: [] (FrozenError)`

**Comment by le0pard** [comment]
Maybe adding `Rails.application.routes_reloader.execute_unless_loaded` in beginning of https://github.com/railsware/js-routes/blob/9bedb7f20a0d125189f8bcd9883d7c21314b47e3/lib/js_routes/instance.rb#L27 may fix problem. Are you able to test @francescob (create fork and modify it) ?

**Comment by francescob** [comment]
yes, that fixed it. I'm creating a pull request

**Comment by bogdan** [comment]
Fixed in 2.3.0
